subsys/net/l2/ethernet/dsa/dsa.c

```diff
@@ -0,0 +1,138 @@
+/** @file
+ * @brief LLDP related functions
+ */
+
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_dsa);
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include <net/net_core.h>
+#include <net/ethernet.h>
+#include <net/net_mgmt.h>
+#include <net/dsa.h>
+
+/*
+ * RECEIVE HANDLING CODE - ingress (ETH -> DSA slave ports)
+ */
+
+static int dsa_check_iface(struct net_if *iface)
+{
+	if (net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)) {
+		return -ENOENT;
+	}
+
+	if (!((net_eth_get_hw_capabilities(iface) & ETHERNET_DSA_MASTER_PORT) ||
+	    (net_eth_get_hw_capabilities(iface) & ETHERNET_DSA_SLAVE_PORT))) {
+		return -ESRCH;
+	}
+
+	return 0;
+}
+
+
+int dsa_register_recv_callback(struct net_if *iface, net_dsa_recv_cb_t cb)
+{
+	struct ethernet_context *ctx;
+	int ret;
+
+	ret = dsa_check_iface(iface);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ctx = net_if_l2_data(iface);
```

Comment:
you could put this into the if statement below (as well at the struct ethernet_context *ctx )

Reply:
Ok.